packages/flutter_tools/test/general.shard/android/java_test.dart

```diff
@@ -127,6 +134,37 @@ OpenJDK 64-Bit Server VM Zulu19.32+15-CA (build 19.0.2+7, mixed mode, sharing)
         );
         expect(java, isNull);
       });
+
+      testWithoutContext('finds and prefers JDK found at config item "jdk-dir" if it is set', () {
+        const String configuredJdkPath = '/jdk';
+        config.setValue('jdk-dir', configuredJdkPath);
+
+        processManager.addCommand(
+          const FakeCommand(
+            command: <String>['which', 'java'],
+            stdout: '/fake/which/java/path',
+          ),
+        );
+
+        final FakePlatform platformWithJavaHome = FakePlatform(
+          environment: <String, String>{
+            'JAVA_HOME': '/old/jdk'
+          },
+        );
+
+        final Java? java = Java.find(
```

Comment:
Can you actually call `Java.find()` twice in this test, once before `config.setValue('jdk-dir', ...)` and once after, expecting the results are different?

Reply:
Isn't this already implicitly covered by the other tests?

Reply:
We can only rely on other tests verifying the exact setup of this test if we know that now and forever this test and the dependent test are set up in exactly the same way. Also that that other test is never deleted.

In other words, I would say that this test is not completely verifying that `Java.find()` is finding `java.javaHome` via the `Config`.